### PR TITLE
Citus optimize list schedules query

### LINKED ISSUE #5235: Citus optimize list schedules query
### Problem

REST API performance test with citus db shows list schedules (with no additional query params) REST API performs poorly.

### Solution

- Run `explain analyze` to understand the bottleneck of the SQL query
- Optimize the query based on the analysis, split the query into multiple parts if necesasry
- Run performance test to verify

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR optimizes the list schedules REST API for citus

- find matching schedules then get the entity properties and signatures in parallel

**Related issue(s)**:

Fixes #5235 

**Notes for reviewer**:

This changes have little to no performance impact for v1. For citus, we do see better performance since once the schedule ids are found, queries for the entity properties and signatures will only push down to the matching shards.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/schedules.js [hunk 1]
```diff
@@ -230,15 +236,40 @@ const getSchedules = async (req, res) => {
   // get sql filter query, params, order and limit from query filters
   const {filterQuery, params, order, limit} = extractSqlFromScheduleFilters(filters);
   const schedulesQuery = getSchedulesQuery(filterQuery, order, params.length);
+  if (logger.isTraceEnabled()) {
+    logger.trace(`getSchedules query: ${schedulesQuery}, params: ${params}`);
```
Comment (xin-hedera) [inline hedera-mirror-rest/schedules.js:240]: we should move such logs into `pool.queryQuietly` then won't be bothered by low code coverage warnings

**Review by mgoelswirlds — APPROVED** [review]

File: hedera-mirror-rest/schedules.js [hunk 2]
```diff
@@ -230,15 +236,40 @@ const getSchedules = async (req, res) => {
...
+  const entityIds = schedules.map((s) => s.schedule_id);
+  const positions = _.range(1, entityIds.length + 1)
+    .map((i) => `$${i}`)
+    .join(',');
+  const entityQuery = `select ${entityFields} from entity where id in (${positions}) order by id ${order}`;
+  const signatureQuery = `select entity_id, ${transactionSignatureJsonAgg} as signatures
+    from transaction_signature ts
+    where entity_id in (${positions})
+    group by entity_id
+    order by entity_id ${order}`;
```
Comment (steven-sheehy) [inline hedera-mirror-rest/schedules.js:260]: There's a risk the sort order for signatures within an entity is non-deterministic. We should add a sort on consensus_timestamp asc

File: hedera-mirror-rest/schedules.js [hunk 3]
```diff
@@ -61,23 +61,24 @@ const transactionSignatureJsonAgg = `
   ) order by ts.consensus_timestamp)`;
 const getScheduleByIdQuery = `
   select
-    ${commonSelectFields},
+    s.consensus_timestamp,
```
Comment (steven-sheehy) [inline hedera-mirror-rest/schedules.js:64]: I think we should just add the table prefix to `scheduleFields` and `entityFields` and join the two here so we don't repeat the fields for list and by ID.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/schedules.js (hunk 2 above)
Comment (xin-hedera) [reply hedera-mirror-rest/schedules.js:260]: the order is done in `transactionSignatureJsonAgg`

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**